packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/ExportAllTasks.tsx

```diff
@@ -0,0 +1,188 @@
+import useAtmosphere from '../../../../../hooks/useAtmosphere'
+import {integrationSvgLookup} from '../../../../../components/TaskIntegrationMenuItem'
+import {Providers} from '../../../../../types/constEnums'
+import GitHubSVG from '../../../../../components/GitHubSVG'
+import JiraSVG from '../../../../../components/JiraSVG'
+import GitLabSVG from '../../../../../components/GitLabSVG'
+import clsx from 'clsx'
+import SendClientSegmentEventMutation from '../../../../../mutations/SendClientSegmentEventMutation'
+
+const ExportAllTasksMenuRoot = lazyPreload(
+  () => import(/* webpackChunkName: 'ExportAllTasksMenuRoot' */ './ExportAllTasksMenuRoot')
+)
+
+const BUTTON_CLASSES =
+  'flex items-center gap-2 rounded-full border border-solid border-slate-400 px-5 py-2 text-center font-sans text-sm font-semibold'
+
+const integrationToServiceName: Record<TaskServiceEnum, string> = {
+  PARABOL: 'Parabol',
+  github: Providers.GITHUB_NAME,
+  jira: 'Jira',
+  jiraServer: Providers.JIRA_SERVER_NAME,
+  azureDevOps: Providers.AZUREDEVOPS_NAME,
+  gitlab: Providers.GITHUB_NAME
+}
+
+interface Props {
+  meetingRef: ExportAllTasks_meeting$key
+}
+
+const ExportAllTasks = (props: Props) => {
+  const {meetingRef} = props
+  const mutationProps = useMutationProps()
+  const atmosphere = useAtmosphere()
+  const {onCompleted, onError, submitMutation, submitting} = mutationProps
+  const {togglePortal, originRef, menuPortal, menuProps, loadingWidth, loadingDelay} = useMenu(
+    MenuPosition.UPPER_RIGHT,
+    {
+      loadingWidth: 200
+    }
+  )
+  const meeting = useFragment(
+    graphql`
+      fragment ExportAllTasks_meeting on NewMeeting {
+        id
+        meetingType
+        teamId
+        ...ExportAllTasksMenuRoot_meeting
+        ... on RetrospectiveMeeting {
+          tasks {
+            id
+            taskService
+            integration {
+              id
+            }
+          }
+        }
+        ... on ActionMeeting {
+          tasks {
+            id
+            taskService
+            integration {
+              id
+            }
+          }
+        }
+      }
+    `,
+    meetingRef
+  )
+
+  // :TODO: (jmtaber129): Get this information from the tasks integration.
+  const [pushedIntegrationLabel, setPushedIntegrationLabel] = useState<string | null>(null)
+
+  const {id: meetingId, meetingType, tasks, teamId} = meeting
+  if (!tasks) {
+    return null
+  }
+
+  const filteredTasks = tasks.filter((task) => !task.integration)
+  const taskServices = Array.from(
+    tasks.reduce(
+      (serviceSet, task) => (task.taskService ? serviceSet.add(task.taskService) : serviceSet),
+      new Set<string>()
+    )
+  )
+
+  const handlePushToIntegration = (
+    integrationRepoId: string,
+    integrationProviderService: Exclude<TaskServiceEnum, 'PARABOL'>,
+    integrationLabel?: string
+  ) => {
+    submitMutation()
+    Promise.allSettled(
```

Comment:
clarifying questions before I continue with the review
1. why client side? this should be done on server, unless I'm missing something
2. if this has to be done on client side, what about hitting rate limiter?

Reply:
1. You're right; this should be moved to the server in the long term. However, leaving this on the frontend seems like a reasonable 80/20 to avoid a large refactor of the `createTaskIntegration` mutation right now (either to modify that mutation to accept multiple task IDs or refactoring the code to be reused by a new bulk mutation). 
2. Looking at our server-side code, this mutation doesn't seem to be protected by a ratelimiter. There's a potential for hitting rate limiters in downstream integrations, but this is an edge case I'm fine leaving unaddressed for now. The code is structured such that if a few tasks fail to push due to e.g. a downstream rate-limiting error, the sync could be retried easily.

Overall, since this is a semi-experimental new feature, I'd prefer to put this into the wild and see that it gets some usage before investing additional effort into making this more robust. If we find users are using this and getting value from it, but it has stability issues, that's when I'd want to invest effort into the server-side refactor. But if this is unstable and gets no usage I'd rather just tear it out than make it robust.

Reply:
Jira's rate limiter for reading issues surprised me with how low it is. It's lower than they say in their docs & it's the reason we have to store all jira issues in redis-- we would often hit it during poker meetings, and that was for _reads_!

I don't see a rate limit for creating issues (https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issues/#api-rest-api-3-issue-post) but they do have a bulk-create API, which leads me to believe one exists.

can you run a stress test & see how many issues it can push before it breaks?

Reply:
I was able to sync 20 tasks without issue, but not 22 tasks. However, the user can simply retry the sync and any unsynced tasks will sync correctly.

I also realized when stress testing that this actually just fails silently, so I also added a snackbar when this fails. There's probably a better failure case UX, but I think this is fine for now:
<img width="482" alt="Screen Shot 2023-05-19 at 2 49 00 PM" src="https://github.com/ParabolInc/parabol/assets/9013217/930fc4aa-0e6d-44df-8266-96c1e6ec41b8">